lib/human_player.rb

```diff
@@ -2,20 +2,24 @@
 
 class HumanPlayer < Player
 
-  def initialize(player_num, board)
+  def initialize(player_num, playing_computer)
     @player_num = player_num
-    @board = board
+    @playing_computer = playing_computer
   end
 
-  def get_next_move
-    IOHandler.new.get_valid_move(@board.to_string_array)
+  def get_next_move(board)
+    IOHandler.new.get_valid_move(board.to_string_array)
   end
 
   def get_token
-    if @player_num == 1
+    if @playing_computer
```

Comment:
Why does it matter if the opponent is the computer?  How is that relevant to which token the player is represented by?

Reply:
This is a calculation on two fields that never change, but you recalculate the result each time `get_token` is called.  This should be calculated either in the constructor or externally and passed to the constructor.  `get_token` can be a simple accessor.

Reply:
This expression is far too complicated.  There is no need for nested `if`s.

Reply:
Token is now passed in, and get_token is just an accessor.